runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.h

```diff
@@ -43,50 +36,155 @@ void configureHeader();
 void insertNoOpCommand(unsigned int numPadBytes);
 }
 
-#define OSTREAM_OP(O_TYPE, TYPE) O_TYPE &operator<<(O_TYPE &os, const TYPE &s);
-
 namespace mlir::iree_compiler::AMDAIE {
-#define TO_STRING(TYPE) std::string to_string(const TYPE &t);
+struct TileLoc {
+  int col, row;
+  inline bool operator<(const TileLoc& rhs) const {
+    return std::tie(col, row) < std::tie(rhs.col, rhs.row);
+  }
+
+  bool operator==(const TileLoc& rhs) const {
+    return std::tie(col, row) == std::tie(rhs.col, rhs.row);
+  }
+
+  bool operator!=(const TileLoc& rhs) const { return !(*this == rhs); }
+
+  operator XAie_LocType() const { return XAie_TileLoc(col, row); }
+  TileLoc(XAie_LocType loc) : col(loc.Col), row(loc.Row) {}
+  TileLoc(int col, int row) : col(col), row(row) {}
+};
+
+enum class AMDAIEDevice : uint32_t {
+  xcvc1902 = 1,
+  xcve2302 = 2,
+  xcve2802 = 3,
+  npu = 4,
+  npu1 = 4,
+  npu1_1col = 5,
+  npu1_2col = 6,
+  npu1_3col = 7,
+  npu1_4col = 8,
+};
+
+enum class AMDAIETileType : uint8_t {
+  AIETILE = 0U,
```

Comment:
The C++ spec says the values increment from 1 after the first one by default, you don't need to enumerate all values here

Reply:
Even for `class`? Still I prefer the explicit enumeration because e.g. for 

```
typedef enum{
	CORE,
	DMA,
	CTRL,
	FIFO,
	SOUTH,
	WEST,
	NORTH,
	EAST,
	TRACE,
	UCTRLR,
	SS_PORT_TYPE_MAX
} StrmSwPortType;
```

I am constantly counting on my numbers (not really) when the enum gets serialized to an integer.

Reply:
Now I'm not sure actually, maybe the spec doesn't say increment by default (sorry for the noise). As for serializing though, shouldn't you always be going with static_cast<int>(AMDAIETileType)(AIETILE), i.e. you never actually need to use an raw integer? I guess for debugging it can be useful though.

Reply:
`printf`s and the like serialize to integers (well at least they did before I implementd `operator<<` all over the place...).